## 06_check_snmp_win by Jan Wagner <waja@cyconet.org>
##
## DP: Update script to version 1.1 

diff -Nur nagios-snmp-plugins-1.1.1.orig/check_snmp_win.pl nagios-snmp-plugins-1.1.1/check_snmp_win.pl
--- nagios-snmp-plugins-1.1.1.orig/check_snmp_win.pl	2006-11-29 20:59:55.000000000 +0100
+++ nagios-snmp-plugins-1.1.1/check_snmp_win.pl	2007-10-12 22:23:25.000000000 +0200
@@ -1,10 +1,11 @@
 #!/usr/bin/perl -w
 ############################## check_snmp_win ##############
-# Version : 0.6
-# Date : Nov 29 2006 
+my $Version='1.1';
+# Date : Oct 12 2007
 # Author  : Patrick Proy (patrick at proy.org)
-# Help : http://www.manubulon.com/nagios/
+# Help : http://nagios.manubulon.com/
 # Licence : GPL - http://www.fsf.org/licenses/gpl.txt
+# Contrib : Tenaku
 # TODO : 
 ###############################################################
 #
@@ -16,10 +17,10 @@
 
 # Nagios specific
 
-use lib "/usr/local/nagios/libexec";
-use utils qw(%ERRORS $TIMEOUT);
-#my $TIMEOUT = 5;
-#my %ERRORS=('OK'=>0,'WARNING'=>1,'CRITICAL'=>2,'UNKNOWN'=>3,'DEPENDENT'=>4);
+#use lib "/usr/local/nagios/libexec";
+#use utils qw(%ERRORS $TIMEOUT);
+my $TIMEOUT = 15;
+my %ERRORS=('OK'=>0,'WARNING'=>1,'CRITICAL'=>2,'UNKNOWN'=>3,'DEPENDENT'=>4);
 
 # SNMP Datas for processes (MIB II)
 my $process_table= '1.3.6.1.2.1.25.4.2.1';
@@ -44,7 +45,6 @@
 
 # Globals
 
-my $Version='0.6';
 my $Name='check_snmp_win';
 
 my $o_host = 	undef; 		# hostname 
@@ -93,7 +93,7 @@
 
 sub help {
    print "\nSNMP Windows Monitor for Nagios version ",$Version,"\n";
-   print "GPL licence, (c)2004-2005 Patrick Proy\n\n";
+   print "GPL licence, (c)2004-2007 Patrick Proy\n\n";
    print_usage();
    print <<EOT;
 -v, --verbose
@@ -344,7 +344,11 @@
 foreach my $List (@o_descrL) {
   my $test=0;
   for (my $i=0; $i< $num_int; $i++) {
-    if ( $descr[$i] =~ /$List/i  ) { $test++; }
+    if (defined($o_noreg)){
+      if ($descr[$i] eq $List) { $test++;}
+    } else {
+      if ( $descr[$i] =~ /$List/i ) { $test++; }
+    }
   }
   if ($test==0) {
     $output .= ", " if defined($output);
